mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -129,13 +133,24 @@ bool MemRefDependenceGraph::init() {
       for (auto *opInst : collector.loadOpInsts) {
         node.loads.push_back(opInst);
         auto memref = cast<AffineReadOpInterface>(opInst).getMemRef();
+        if (memrefAccesses.find(memref) == memrefAccesses.end()) {
```

Comment:
When a memory reference is accessed for the first time, all the nodes with possible memory effects that precede it are added to its access list.